binder/views.py

```diff
@@ -1802,20 +1805,21 @@ def post(self, request, pk=None):
 
 
 
-	def delete(self, request, pk=None, undelete=False):
+	def delete(self, request, pk=None, undelete=False, skip_body_check=False):
 		if not undelete:
 			self._require_model_perm('delete', request)
 
 		if pk is None:
 			raise BinderMethodNotAllowed()
 
-		# FIXME: ugly workaround, remove when Django bug fixed
-		# Try/except because https://code.djangoproject.com/ticket/27005
-		try:
-			if request.body not in (b'', b'{}'):
-				raise BinderRequestError('{}DELETE body must be empty or empty json object.'.format('UN' if undelete else ''))
-		except ValueError:
-			pass
+		if not skip_body_check:
```

Comment:
Just for my information, how would this affect existing code? It is fully backwards compatible. Or am I missing something?

I do agree with the fact that it is not really a nice change. I did consider the change you proposed as well during implementation, but I could not defend the implicit logic you are adding to the function. Therefore I took the approach to be explicit rather than implicit about what the function is doing.

So, what doesn't make sense to me is that with your proposal you are allowed to DELETE with an non-empty body if you do a PUT/PATCH request, but not if you do a DELETE request. If you would call this function somewhere else in your application, (e.g. a custom endpoint) this wouldn't make sense at all, and can cause weird bugs.

The real problem here is that there is not the correct level of abstraction in the Binder views, and that I can't delete a model with all the logic attached to it, without "mocking" a delete request. I don't think we are going to solve this problem now though, so apart of your and mine solution, there are two other solutions we might consider:

- Remove the empty body check all together. If I understand correctly, this is just a sanity check which warns the frontenders that they are doing something wrong
- Move this code to the dispatch method, which then attaches this logic to the actual "DELETE" http method. This will have some major consequences for some of our projects which use DELETE endpoints, with logic depending on the body. (E.g. deleting a list of objects)

Maybe @Daanvdk Can give his two cents about it

Reply:
> Just for my information, how would this affect existing code? It is fully backwards compatible. Or am I missing something?

It wouldn't be if someone had copy/paste overridden the `delete` method (which wouldn't accept that new keyword).

But I fully agree the abstraction is all wrong here.